app/models/concerns/abilities/task_ability.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Abilities
+module Abilities
+  # Abilities on task
+  module TaskAbility
+    def define_task_abilities_for_admin(user)
+      can :manage, Task, company_id: user.company_id
+    end
+
+    def define_task_abilities_for_employee(user)
+      can %i[read create], Task
+      can :update, Task do |task|
+        task.try(:user) == user
+      end
+      can :destroy, Task do |task|
+        task.try(:user) == user
```

Comment:
can [:update], Task, created_by_id: user.id, company_id: user.id

need to handle project id as well but in a separate PR when project PR will be merged